src/Compilers/CSharp/Test/Semantic/Semantics/NonTrailingNamedArgumentsTests.cs

```diff
@@ -0,0 +1,418 @@
+	public MyAttribute(bool condition, int other) { }
+}
+
+[MyAttribute(condition: true, 42)]
+[MyAttribute(condition: true, P = 1, 42)]
+[MyAttribute(42, condition: true)]
+public class C
+{
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular7_2);
+            comp.VerifyDiagnostics(
+                // (12,38): error CS1016: Named attribute argument expected
+                // [MyAttribute(condition: true, P = 1, 42)]
+                Diagnostic(ErrorCode.ERR_NamedArgumentExpected, "42").WithLocation(12, 38),
+                // (13,18): error CS1744: Named argument 'condition' specifies a parameter for which a positional argument has already been given
+                // [MyAttribute(42, condition: true)]
+                Diagnostic(ErrorCode.ERR_NamedArgumentUsedInPositional, "condition").WithArguments("condition").WithLocation(13, 18)
+                );
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+            var nodes = tree.GetCompilationUnitRoot().DescendantNodes();
+            var invocation = nodes.OfType<AttributeSyntax>().ElementAt(1);
+            Assert.Equal("MyAttribute(condition: true, 42)", invocation.ToString());
+            Assert.Equal("MyAttribute..ctor(System.Boolean condition, System.Int32 other)",
+                model.GetSymbolInfo(invocation).Symbol.ToTestDisplayString());
+        }
+
+        [Fact]
+        public void TestInAttribute2()
+        {
+            var source = @"
+using System;
+
+[AttributeUsage(AttributeTargets.All, AllowMultiple = true)]
+public class MyAttribute : Attribute
+{
+    public int P { get; set; }
+	public MyAttribute(int a = 1, int b = 2, int c = 3) { }
+}
+
+[MyAttribute(c:3, 2)]
+[MyAttribute(P=1, c:3, 2)]
+public class C
+{
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular7_2);
+            comp.VerifyDiagnostics(
+                // (11,14): error CS8321: Named argument 'c' is used out-of-position but is followed by an unnamed argument
+                // [MyAttribute(c:3, 2)]
+                Diagnostic(ErrorCode.ERR_BadNonTrailingNamedArgument, "c").WithArguments("c").WithLocation(11, 14),
+                // (12,21): error CS1016: Named attribute argument expected
+                // [MyAttribute(P=1, c:3, 2)]
+                Diagnostic(ErrorCode.ERR_NamedArgumentExpected, "3").WithLocation(12, 21),
+                // (12,24): error CS1016: Named attribute argument expected
+                // [MyAttribute(P=1, c:3, 2)]
+                Diagnostic(ErrorCode.ERR_NamedArgumentExpected, "2").WithLocation(12, 24),
+                // (12,19): error CS8321: Named argument 'c' is used out-of-position but is followed by an unnamed argument
+                // [MyAttribute(P=1, c:3, 2)]
+                Diagnostic(ErrorCode.ERR_BadNonTrailingNamedArgument, "c").WithArguments("c").WithLocation(12, 19)
```

Comment:
> Diagnostic(ErrorCode.ERR_BadNonTrailingNamedArgument, "c").WithArguments("c").WithLocation(12, 19) [](start = 15, length = 99)

This error feels misleading, we already reported ERR_NamedArgumentExpected and providing the name for the next argument wouldn't fix the problem. 
Right now we report two CS1016 errors for this line and do not complain about named argument in the middle at all.